### PR TITLE
fixed the race condition in ConsistentHashingRouterSpec

### PR DESCRIPTION
**Pull request by annymsMthd**
ConsistentHashingRouterMustAdjustNodeRingWhenRouteeDies

Seems like removing the ask and task surrounding the test fixes the issue. Need another set of eyes on it to ensure I'm not hiding a deeper issue.

### CODE REVIEW

**Comment by annymsMthd** [comment]
#853 issue link

**Comment by Aaronontheweb** [comment]
PR passed build on the first try - going to run it a few times to confirm.

**Comment by annymsMthd** [comment]
Awesome. Yeah keep hitting it I want to see what other ones are lurking around.

File: src/core/Akka/Routing/TailChoppingRoutingLogic.cs [hunk 1]
```diff
@@ -119,27 +119,33 @@ public override void Send(object message, IActorRef sender)
...
 
-            _scheduler.Advanced.ScheduleOnce(_within, () => 
+            if (_routees.Length == 0)
             {
-                completion.TrySetException(new TimeoutException(String.Format("Ask timed out on {0} after {1}", sender, _within)));
-            }, cancelable);
-
-            var request = completion.Task;
-            completion.Task.ContinueWith(task => 
+                completion.TrySetResult(NoRoutee);
```
Comment (annymsMthd) [inline src/core/Akka/Routing/TailChoppingRoutingLogic.cs]: Would this be the appropriate message to send back if there are no routees?

Comment (Aaronontheweb) [inline src/core/Akka/Routing/TailChoppingRoutingLogic.cs]: Yes

**Comment by Aaronontheweb** [comment]
@annymsMthd looks good - could you rebase on dev? PR we pulled in this morning was a big change (memory footprint optimization)

**Comment by annymsMthd** [comment]
I think i might have messed up the rebase. Does this look right?

**Comment by annymsMthd** [comment]
Closing this because i messed up on the rebase

**Pull request closed without merging**